File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TranslationHandler.java
package com.abubusoft.xenon.animations;

import com.abubusoft.xenon.math.Vector3;

public class TranslationHandler extends AnimationHandler<TranslationFrame> {
	@Override
	public TranslationFrame value(TranslationFrame current, long enlapsedTime, TranslationFrame next) {
		if (next != null) {
			float perc = current.interpolation.getPercentage(enlapsedTime, current.duration * this.rate);

			// temp viene sempre scritto, non importa cosa c'è prima
			Vector3.multiply(next.translation, perc, temp.translation);
			return temp;
		} else {
			// siamo sull'ultimo frame, dopo non c'è niente. La traslazione è 0
			temp.translation.setCoords(0, 0, 0);
			return temp;
		}
	}

	@Override
	public TranslationFrame buildFrame() {
		return new TranslationFrame();
	}
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TranslationHandler.kt
package com.abubusoft.xenon.animations

import com.abubusoft.xenon.math.Vector3

class TranslationHandler : AnimationHandler<TranslationFrame?>() {
    public override fun value(current: TranslationFrame?, enlapsedTime: Long, next: TranslationFrame?): TranslationFrame? {
        return if (next != null) {
            val perc = current!!.interpolation.getPercentage(enlapsedTime.toFloat(), current.duration * rate)

            // temp viene sempre scritto, non importa cosa c'è prima
            Vector3.multiply(next.translation, perc, temp!!.translation)
            temp
        } else {
            // siamo sull'ultimo frame, dopo non c'è niente. La traslazione è 0
            temp!!.translation.setCoords(0f, 0f, 0f)
            temp
        }
    }

    public override fun buildFrame(): TranslationFrame? {
        return TranslationFrame()
    }
}